lib/isexpired.c

```diff
@@ -22,6 +22,24 @@
 
 #ident "$Id$"
 
+/*
+ * addsl - adds three non-negative long values
+ *
+ *	the result is capped at LONG_MAX.
+ */
+static long addsl(long v1, long v2, long v3)
+{
+	long value;
+
+	if (v1 > LONG_MAX - v2)
+		return LONG_MAX;
+	value = v1 + v2;
+
+	if (value > LONG_MAX - v3)
+		return LONG_MAX;
+
+	return value + v3;
+}
```

Comment:
We could also make it slightly more generic to accept negative numbers and saturate to LONG_MIN, although that would increase complexity a little bit.

Reply:
Since the input values are always positive, this adds complexity without benefit.

Reply:
Agree.  After implementing it myself, it proves to be quite complex.

The benefit would be that if someone uses that function in a case where we don't know if the values are all non-negative, we may get surprises, but if the function is prepared to receive negative values, then it's all fine.

I'll try to add it myself in a separate patch, and provide tests for it.